plugins/coordinator/plugin.go

```diff
@@ -123,24 +123,20 @@ func handleError(err error) bool {
 		return false
 	}
 
-	var critErr common.CriticalError
-	var softErr common.SoftError
-
-	switch {
-	case errors.As(err, &critErr):
+	if err := common.IsCriticalError(err); err != nil {
```

Comment:
should we shadow the original `err` here?
Shouldn't be a problem, just mentioning it :)